lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3470,8 +3486,9 @@ namespace {
 
       result->setHasUnreferenceableStorage(hasUnreferenceableStorage);
 
-      if (auto cxxRecordDecl = dyn_cast<clang::CXXRecordDecl>(decl)) {
-        result->setIsCxxNonTrivial(!cxxRecordDecl->isTriviallyCopyable());
+      if (cxxRecordDecl) {
+        result->setIsCxxNonTrivial(
+            !cxxRecordDecl->isTriviallyCopyable());
```

Comment:
This is unrelated to this PR, but are we checking properly for non-copyable types?

Reply:
I'm not 100% sure. We'll need to look into this more deeply in #32378.